crates/quick_action_bar/src/quick_action_bar.rs

```diff
@@ -121,12 +122,17 @@ impl Render for QuickActionBar {
             },
         );
 
-        h_flex()
+        let mut bar = h_flex()
             .id("quick action bar")
             .gap_2()
             .children(inlay_hints_button)
-            .children(search_button)
-            .child(assistant_button)
+            .children(search_button);
+
+        if AssistantSettings::get_global(cx).button {
+            bar = bar.child(assistant_button);
```

Comment:
Instead of using this mutable assignment we can use when of the alternative APIs (like `.when`) to conditionally add the button based on the setting:

```rs
h_flex()
    .id("quick action bar")
    .gap_2()
    .children(inlay_hints_button)
    .children(search_button)
    .when(AssistantSettings::get_global(cx).button, |bar| {
        bar.child(assistant_button)
    })
```